numpy/polynomial/_polybase.py

```diff
@@ -284,10 +296,12 @@ class as self with identical domain and window. If so,
                 raise TypeError("Domains differ")
             elif not np.all(self.window == other.window):
                 raise TypeError("Windows differ")
+            elif self.symbol != other.symbol:
+                raise ValueError("Polynomial symbols differ")
```

Comment:
Should here be some guidance on renaming the symbol?  Something like: `to change a  polynomial's symbol, use p.symbol = 't'` (or whatever the right way is).